awx/main/management/commands/host_metrics.py

```diff
@@ -0,0 +1,69 @@
+from django.core.management.base import BaseCommand
+from dateutil import parser
+from django.utils.timezone import now
+import pytz
+from datetime import datetime, timezone
+from awx.main.models.inventory import HostMetrics
+import json
+
+
+class Command(BaseCommand):
+
+    help = 'This is for offline licensing usage'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--since', dest='since', action='store', help='Start date for data')
+        parser.add_argument('--until', dest='until', action='store', help='End date for data')
+        parser.add_argument('--json', action='store_true', help='Select output as JSON')
+
+    def handle(self, *args, **options):
+        opt_since = options.get('since')
+        opt_until = options.get('until')
+
+        # user gives no arguments
+        if opt_since is None and opt_until is None:
+            print("No Arguments received")
+            return None
+
+        since = None
+        if opt_since:
+            since = parser.parse(opt_since)
+            if since.tzinfo is None:
+                since = since.replace(tzinfo=timezone.utc)
+
+        until = None
+        if opt_until:
+            until = parser.parse(opt_until)
+            if until.tzinfo is None:
+                until = until.replace(tzinfo=timezone.utc)
+
+        if (since is None) and (until is not None):
+            filter_kwargs = {"last_automation__lte": until}
+        elif (since is not None) and (until is None):
+            filter_kwargs = {"last_automation__gte": since}
+        else:
+            filter_kwargs = {"last_automation__gte": since, "last_automation__lte": until}
+
+        result = HostMetrics.objects.filter(**filter_kwargs)
+
+        # if --json flag is set, output the result in json format
+        if options['json']:
+            list_of_queryset = list(result.values('hostname', 'first_automation', 'last_automation'))
+            for item in list_of_queryset:
+                item['first_automation'] = item['first_automation'].strftime('%Y-%m-%d')
```

Comment:
If people run this on a weeks/months timeframe, this could end being a *very* large dataset to keep around in memory.

@wenottingham do you *care* about printing a full audit of *all* the possible automations (in tabular format), or do you want just a *total number* printed.

If we want this ability, maybe we could print the full list with an _optional_ flag to the CLI, and by default just show the count?

Reply:
A customer may need to/want to reconcile certain hosts that are "decommissioned on January 31st" vs just "not automated on since January 31st"

To do that, we would need to dump the full list. It doesn't need to be the default, but it does need to at least be an option.